extensions/data-plane/data-plane-public-api-v2/src/test/java/org/eclipse/edc/connector/dataplane/api/controller/DataPlanePublicApiV2ControllerTest.java

```diff
@@ -117,7 +117,24 @@ void should_returnInternalServerError_if_transferFails() {
                 .then()
                 .statusCode(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode())
                 .contentType(JSON)
-                .body("errors[0]", is(GENERAL_ERROR + ": " + errorMsg));
+                .body("errors[0]", is(errorMsg));
+    }
+    @Test
+    void should_returnListOfErrorsAsAResponse_if_anythingFails() {
+        var token = UUID.randomUUID().toString();
+        var errorMsg = UUID.randomUUID().toString();
+        when(dataAddressResolver.resolve(any())).thenReturn(Result.success(testDestAddress()));
+        when(pipelineService.transfer(any(), any()))
+                .thenReturn(completedFuture(StreamResult.error(errorMsg)));
+
+        baseRequest()
+                .header(AUTHORIZATION, token)
+                .when()
+                .post("/any")
+                .then()
+                .statusCode(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode())
+                .contentType(JSON)
+                .body("errors", is(List.class));
```

Comment:
This test does not properly verify the change. Compare it to how the original test verifies the error messages.